odm/firestore_abstract_repository.go

```diff
@@ -0,0 +1,252 @@
+package odm
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"cloud.google.com/go/firestore"
+	"github.com/SaiNageswarS/go-api-boot/logger"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.uber.org/zap"
+	"google.golang.org/api/iterator"
+)
+
+type FirestoreBootRepository[T any] interface {

```

Comment:
Don't create another interface as suggested in overview comments. Let there be one BootRepository Interface and multiple Unimplemented{DB}Repositories.